server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -1718,10 +1744,30 @@ private void closeUnsealedTsFileProcessorCallBack(
   /**
    * close compaction merge callback, to release some locks
    */
-  private void closeCompactionMergeCallBack() {
-    this.compactionMergeWorking = false;
-    synchronized (closeStorageGroupCondition) {
-      closeStorageGroupCondition.notifyAll();
+  private void closeCompactionMergeCallBack(boolean isMerge, long timePartitionId) {
+    if (isMerge && IoTDBDescriptor.getInstance().getConfig().isEnableContinuousCompaction()) {
+      if (!CompactionMergeTaskPoolManager.getInstance().isTerminated()) {
```

Comment:
this if-else could be removed as during the whole process, `CompactionMergeTaskPoolManager` is never Terminated

Reply:
When IoTDB is closing it can be terminated, and we should not keep compaction